presto-hive/src/main/java/com/facebook/presto/hive/HiveTableHandle.java

```diff
@@ -30,20 +30,31 @@
 {
     private final String schemaName;
     private final String tableName;
+    private final Optional<Long> snapshotTimestampMS;
 
     private final Optional<List<List<String>>> analyzePartitionValues;
 
     @JsonCreator
     public HiveTableHandle(
             @JsonProperty("schemaName") String schemaName,
             @JsonProperty("tableName") String tableName,
+            @JsonProperty("snapshotTimestampMS") Optional<Long> snapshotTimestampMS,
```

Comment:
If we want to add this to `HiveTableHandle`, we probably should at least do some research on OSS Hive metastore and hook it up properly if snapshot if supported. Otherwise, if we want to do a quick prototype, which I think is closer to the actual scope right now, we probably should implement this in `PrismTableHandle` instead, which is a Facebook specific fork of Hive, with support to Facebook infrastructure.